src/apps/intel_mp/intel_mp.lua

```diff
@@ -1348,17 +1330,34 @@ function Intel82599:init ()
    self:unlock_sw_sem()
 end
 
--- helper method for checking that the main process used the same
--- VMDq setting if this is a worker process (noop on main)
+-- Implements various status checks related to VMDq configuration.
+-- Also checks that the main process used the same VMDq setting if
+-- this is a worker process
 function Intel82599:check_vmdq ()
-   if not self.master then
-      if self.vmdq then
-         assert(self.r.MRQC:bits(0, 4) == self.mrqc_bits,
-                "VMDq not set by the main process for this NIC")
-      else
+   if not self.vmdq then
+      assert(not self.macaddr, "VMDq must be set to use MAC address")
+      assert(not self.mirror, "VMDq must be set to specify mirroring rules")
+
+      if not self.master then
          assert(self.r.MRQC:bits(0, 4) ~= self.mrqc_bits,
                 "VMDq was set by the main process for this NIC")
       end
+   else
+      assert(self.driver == "Intel82599", "VMDq only supported on 82599")
+      assert(self.macaddr, "MAC address must be set in VMDq mode")
+
+      if self.driver == "Intel82599" then
```

Comment:
Given the assert of `self.driver == "Intel82599"` three lines above, does this check make sense?